yt_dlp/extractor/viu.py

```diff
@@ -312,35 +295,50 @@ def _real_extract(self, url):
             'ccs_product_id': video_data['ccs_product_id'],
             'language_flag_id': self._LANGUAGE_FLAG.get(lang_code.lower()) or '3',
         }
+
+        rand = ''.join(random.choice('0123456789') for _ in range(10))
+        bearer_token = self._download_json(
+            f'https://api-gateway-global.viu.com/api/auth/token?v={rand}000',
+            video_id=None,
+            data=json.dumps({
+                'countryCode': country_code.upper(),
+                'platform': 'browser',
+                'platformFlagLabel': 'web',
+                'language': 'en',
+                'uuid': str(uuid.uuid4()),
+                'carrierId': '0'
+            }).encode('utf-8'),
+            headers={'Content-Type': 'application/json'},
+            note='Getting bearer token')['token']
```

Comment:
put this in _real_initialize

Reply:
> ~put this in _real_initialize~ cache the token in memory

Ah ok later after my lunch

Reply:
I'm back. I'll update it now

Reply:
Wait a minute I'm checking on something

Reply:
@pukkandan Could you check?

Reply:
@pukkandan can we just revert for now without caching the token?
I don't know yet how to parse the URL argument inside the _real__initialize to get the country code

Reply:
Which is why I edited my comment...
You can cache token in real_extract itself:

```py
auth_codes = {}
def _real_extract(self, url):
    ...
    country_code = ...
    if not self.auth_codes.get(country_code):
        self.auth_codes[country_code] = ...
    ...
```

Reply:
> Which is why I edited my comment... You can cache token in real_extract itself:
> 
> ```python
> auth_codes = {}
> def _real_extract(self, url):
>     ...
>     country_code = ...
>     if not self.auth_codes.get(country_code):
>         self.auth_codes[country_code] = ...
>     ...
> ```

Thanks.

Reply:
@pukkandan it's now updated. For MY and ID it's not possible for now. Their API is different and should have separated IE Class for that.